src/torcontrol.cpp

```diff
@@ -368,14 +368,26 @@ void TorController::add_onion_cb(TorControlConnection& _conn, const TorControlRe
     }
 }
 
+static bool AllowOutboundOnion()
+{
+    if (!gArgs.IsArgSet("-onlynet")) {
+        return true;
+    }
+    for (const auto& net : gArgs.GetArgs("-onlynet")) {
+        if (ParseNetwork(net) == NET_ONION) {
+            return true;
+        }
+    }
+    return false;
+}
+
 void TorController::auth_cb(TorControlConnection& _conn, const TorControlReply& reply)
 {
     if (reply.code == 250) {
         LogPrint(BCLog::TOR, "tor: Authentication successful\n");
 
-        // Now that we know Tor is running setup the proxy for onion addresses
-        // if -onion isn't set to something else.
-        if (gArgs.GetArg("-onion", "") == "") {
+        // Now that we know Tor is running, possibly set the proxy for onion addresses.
+        if (gArgs.GetArg("-onion", "") == "" && AllowOutboundOnion()) {
```

Comment:
```suggestion
        if (gArgs.GetArg("-onion", "").empty() && AllowOutboundOnion()) {
```

Reply:
You suggested this one earlier and I deliberately skipped it but did not explain why.

Here we are testing whether the result of `GetArg()` equals to its second argument, so it is better to keep that obvious, e.g. `GetArg(..., X) == X`.

Reply:
Because `==` will generate/call string comparison code, instead of just checking string length using `empty()`. Better just use specialized member functions instead of some kind "generic" comparison.

Another example is using `clear()` vs `= ""`: https://www.youtube.com/watch?v=3X9qK7HWxjk

Reply:
I think readability trumps micro-optimization here, leaving it as is.